Gordon360/Services/RecIM/MatchService.cs

```diff
@@ -0,0 +1,217 @@
+                                            OpposingScore = matchTeamJoin.OpposingScore,
+                                            Status = matchTeamJoin.Status,
+                                            Time = match.Time
+                                        })
+                                    .OrderByDescending(history => history.Time)
+                                    .Take(5),
+                                TeamRecord = _context.SeriesTeam
+                                           .Where(st => st.SeriesID == m.SeriesID && st.TeamID == mt.TeamID)
+                                           .Select(st => new TeamRecordViewModel
+                                           {
+                                               Win = st.Win,
+                                               Loss = st.Loss ?? 0,
+                                           })
+                            })
+                        }).FirstOrDefault();
+            return match; 
+        }
+        public IEnumerable<MatchViewModel> GetMatchBySeriesID(int seriesID)
+        {
+            var match = _context.Match
+                        .Where(m => m.SeriesID == seriesID)
+                        .Select(m => new MatchViewModel
+                        {
+                            ID = m.ID,
+                            Time = m.Time,
+                            Surface = _context.Surface
+                                        .FirstOrDefault(s => s.ID == m.SurfaceID)
+                                        .Description,
+                            Status = _context.MatchStatus
+                                        .FirstOrDefault(ms => ms.ID == m.StatusID)
+                                        .Description,
+                            Team = m.MatchTeam.Select(mt => new TeamViewModel
+                            {
+                                ID = mt.TeamID,
+                                Name = _context.Team
+                                   .FirstOrDefault(t => t.ID == mt.TeamID)
+                                   .Name,
+                                TeamRecord = _context.SeriesTeam
+                                           .Where(st => st.SeriesID == m.SeriesID && st.TeamID == mt.TeamID)
+                                           .Select(st => new TeamRecordViewModel
+                                           {
+                                               Win = st.Win,
+                                               Loss = st.Loss ?? 0,
+                                           })
+                            })
+                        });
+            return match;
+        }
+        public async Task PostMatch(CreateMatchViewModel m)
+        {
+            var match = new Match
+            {
+                SeriesID = m.SeriesID,
+                Time = m.Time,
+                SurfaceID = m.SurfaceID ?? 0, //unknown surface id
```

Comment:
can't SurfaceID is a required field and integers cant be nulled unfortunately and the default would be 0

Reply:
I would suggest that you allow it to be nulled in the database so that you can represent the absence of a relationship. 0 is still technically an ID and I believe that SQL Server will try to validate it in a FK relationship. Even if it wasn't a foreign key, null more commonly represents nothing while 0 just represents the number zero.

You can have a nullable int. That would be created in the DB model if you update the table.